### ISSUE TITLE
DateAxis With UTC in Local out

### ISSUE THREAD

**Issue by mcgodan** [amcharts/amcharts4#2539]
So i am trying to import external data and i would like the incoming date format to be utc and then be displayed in the users local date/time but i can't seem to get it to work, possibly because the date format isn't stored correctly? 

Using the following code a utc date e.g. '2020-06-03 10:52:00' is being displayed as '03-Jun-2020 10:52' instead of '11:52'
`
chart.dateFormatter.inputDateFormat = "yyyy-MM-dd HH:mm";
`
`
chart.dateFormatter.utc = true;
`
`
dateAxis.tooltipDateFormat = "dd-MMM-yyyy HH:mm";
`
if the incoming date needs to have the T and Z to work correctly, is there a way to manipulate this on the way in?

Any help would be appreciated.

**Comment by martynasma** [comment]
Hm, the `utc` setting in `DateFormatter` works both ways. It treats string dates as UTC and it makes it output in UTC, not user's time zone.

Solutions. (these will not work, though until we fix a bug I just found; see note below)

1) Obviously, easiest would be to format your source data to either include timezone information, e.g. `".... +0000"` and use `".... ZZ"` in input format.

2) If that is not an option, you can attach the `"+0000"` to your data in pre-processing:

```
chart.events.on("beforedatavalidated", function(ev) {
  // iterate thorugh chart.data and attach "+0000" to date strings
  // ...
});
```

NOTE: for some reason "ZZ" format indicator does not work. We'll fix in the next version and notify you here.

**Comment by mcgodan** [comment]
Hi marrynasma, thanks for getting back to me, I did try using the beforedatavalidated event as you suggested after I posted here actually. I tried adding in the T and Z as ‘yyyy-MM-ddTHH:mmZ’ without any change, I also tried a new Date(Date.parse(...)) without success also. 

Would these be affected by the same bug?

**Comment by martynasma** [comment]
Nah adding "Z" to dates does not mean anything unless you are using ISO date format (`2017-10-14T05:24:17.872Z`) and special input format (`i`).

You would need to add `+0000`. But we will need to fix this first. Expect an update before end of the week.

**Comment by martynasma** [comment]
Fixed in [4.9.25](https://github.com/amcharts/amcharts4/releases/tag/4.9.25).

## [4.9.25] - 2020-06-06

### Added
- Global option `am4core.options.autoDispose` added (default `false`). If set to `true`, when chart is being created in a container that has already existing chart, instead of "Chart was not disposed" warning in console, the chart being overwritten will be disposed.
- New plugin: [Range Selector](https://www.amcharts.com/docs/v4/tutorials/plugin-range-selector/). Collection of controls for adding alternative zoom controls for axes.

### Changed
- Disabled callbacks will now carry over to a clone when cloning an Adapter.

### Fixed
- Export: Sometimes background was improperly clipped when upscaling exported image.
- Export: If chart's `data` was updated open `ExportMenu` was forcibly being closed.
- `"Z"` and `"ZZ"` indicators in `inputDateFormat` were being ignored.
- If series had `sequencedInterpolation = true` set and it was hidden via legend, its legend item lost disabled color when chart was resized.
- `bullet.locationX` and `bullet.locationY` was being ignored on a `ColumnSeries` with `CategoryAxis`.
- If a chart's Legend was disposed, it could result in JS errors if chart had an `XYCursor` enabled.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
